pkg/build/api/validation/validation.go

```diff
@@ -144,23 +144,22 @@ func validateBuildConfigOutput(output *buildapi.BuildOutput) errs.ValidationErro
 func validateStrategy(strategy *buildapi.BuildStrategy) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
 
-	if len(strategy.Type) == 0 {
+	switch {
+	case len(strategy.Type) == 0:
```

Comment:
Strictly speaking since Strategy has its own type, its zero value (which we already check indirectly for erroring) should be represented somehow as well. It's just an API design matter.